From 394af2cdd1a1ec9cfe3a1c4b45a88ff403e10cfb Mon Sep 17 00:00:00 2001
From: Srinivas Kandagatla <srinivas.kandagatla@linaro.org>
Date: Wed, 9 May 2018 13:56:15 +0100
Subject: [PATCH 255/334] ASoC: qdsp6: dt-bindings: Add q6afe dt bindings

This patch add DT bindings for AFE (Audio Frontend) DSP module.

Signed-off-by: Srinivas Kandagatla <srinivas.kandagatla@linaro.org>
Reviewed-and-tested-by: Rohit kumar <rohitkr@codeaurora.org>
Reviewed-by: Rob Herring <robh@kernel.org>
Reviewed-by: Banajit Goswami <bgoswami@codeaurora.org>
Signed-off-by: Mark Brown <broonie@kernel.org>

Conflicts:
	Documentation/devicetree/bindings/sound/qcom,q6afe.txt
	include/dt-bindings/sound/qcom,q6afe.h
---
 .../devicetree/bindings/sound/qcom,q6afe.txt  | 59 +++++++++----------
 include/dt-bindings/sound/qcom,q6afe.h        |  2 +-
 2 files changed, 29 insertions(+), 32 deletions(-)

diff --git a/Documentation/devicetree/bindings/sound/qcom,q6afe.txt b/Documentation/devicetree/bindings/sound/qcom,q6afe.txt
index 1e4b82e7747a..14335a08b963 100644
--- a/Documentation/devicetree/bindings/sound/qcom,q6afe.txt
+++ b/Documentation/devicetree/bindings/sound/qcom,q6afe.txt
@@ -2,39 +2,43 @@ Qualcomm Audio Front End (Q6AFE) binding
 
 AFE is one of the APR audio service on Q6DSP
 Please refer to qcom,apr.txt for details of the common apr service bindings
-used by all apr services.
-
-- but must contain the following property:
+used by all apr services. Must contain the following properties.
 
 - compatible:
 	Usage: required
 	Value type: <stringlist>
-	Definition: must be "qcom,afe-v<MAJOR-NUMBER>.<MINOR-NUMBER>"
-		  Or "qcom,afe" where the version number can be queried
+	Definition: must be "qcom,q6afe-v<MAJOR-NUMBER>.<MINOR-NUMBER>"
+		  Or "qcom,q6afe" where the version number can be queried
 		  from DSP.
-		  example "qcom,afe-v2.0"
+		  example "qcom,q6afe"
 
 = AFE DAIs (Digial Audio Interface)
-"dais" subnode of the AFE node represents dai specific configuration
+"dais" subnode of the AFE node. It represents afe dais, each afe dai is a
+subnode of "dais" representing board specific dai setup.
+"dais" node should have following properties followed by dai children.
 
 - #sound-dai-cells
 	Usage: required
-	Value type: <prop-encoded-array>
+	Value type: <u32>
 	Definition: Must be 1
 
-- reg
+- #address-cells
 	Usage: required
-	Value type: <prop-encoded-array>
-	Definition: Must be dai id
+	Value type: <u32>
+	Definition: Must be 1
 
-- interface-type
+- #size-cells
 	Usage: required
-	Value type: <stringlist>
-	Definition: Must be one of following
-	"mi2s"
-	"hdmi"
-	"pcm"
-	"slim"
+	Value type: <u32>
+	Definition: Must be 0
+
+== AFE DAI is subnode of "dais" and represent a dai, it includes board specific
+configuration of each dai. Must contain the following properties.
+
+- reg
+	Usage: required
+	Value type: <u32>
+	Definition: Must be dai id
 
 - qcom,sd-lines
 	Usage: required for mi2s interface
@@ -44,64 +48,57 @@ used by all apr services.
 
 = EXAMPLE
 
-q6afe {
+q6afe@4 {
 	compatible = "qcom,q6afe";
-	qcom,apr-svc-name = "AFE";
-	qcom,apr-svc-id = <APR_SVC_AFE>;
+	reg = <APR_SVC_AFE>;
 
-	q6asmdai: dais {
+	dais {
 		#sound-dai-cells = <1>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
 		hdmi@1 {
 			reg = <1>;
-			interface-type = "hdmi";
 		};
 
 		prim-mi2s-rx@16 {
 			reg = <16>;
 			qcom,sd-lines = <1 3>;
-			interface-type = "mi2s";
 		};
 
 		prim-mi2s-tx@17 {
 			reg = <17>;
 			qcom,sd-lines = <2>;
-			interface-type = "mi2s";
 		};
 
 		sec-mi2s-rx@18 {
 			reg = <18>;
 			qcom,sd-lines = <1 4>;
-			interface-type = "mi2s";
 		};
 
 		sec-mi2s-tx@19 {
 			reg = <19>;
 			qcom,sd-lines = <2>;
-			interface-type = "mi2s";
 		};
 
 		tert-mi2s-rx@20 {
 			reg = <20>;
 			qcom,sd-lines = <2 4>;
-			interface-type = "mi2s";
 		};
 
 		tert-mi2s-tx@21 {
 			reg = <21>;
 			qcom,sd-lines = <1>;
-			interface-type = "mi2s";
 		};
 
 		quat-mi2s-rx@22 {
 			reg = <22>;
 			qcom,sd-lines = <1>;
-			interface-type = "mi2s";
 		};
 
 		quat-mi2s-tx@23 {
 			reg = <23>;
 			qcom,sd-lines = <2>;
-			interface-type = "mi2s";
 		};
 	};
 };
diff --git a/include/dt-bindings/sound/qcom,q6afe.h b/include/dt-bindings/sound/qcom,q6afe.h
index 3c7868394889..e162045f5dc9 100644
--- a/include/dt-bindings/sound/qcom,q6afe.h
+++ b/include/dt-bindings/sound/qcom,q6afe.h
@@ -1,4 +1,4 @@
-// SPDX-License-Identifier: GPL-2.0
+/* SPDX-License-Identifier: GPL-2.0 */
 #ifndef __DT_BINDINGS_Q6_AFE_H__
 #define __DT_BINDINGS_Q6_AFE_H__
 
-- 
2.24.1

